packages/core/src/compiler.ts

```diff
@@ -452,12 +452,40 @@ export function makeGloballyAddressable(
 class TrackedBuilder implements Builder {
   readonly dependencies = new Set<string>();
   constructor(private realBuilder: Builder) {}
-  getCompiledCard(url: string): Promise<CompiledCard> {
+
+  async getCompiledCard(url: string): Promise<CompiledCard> {
     this.dependencies.add(url);
-    return this.realBuilder.getCompiledCard(url);
+    let card: CompiledCard | undefined;
+    await this.captureMissingCardDependencies(async () => {
+      card = await this.realBuilder.getCompiledCard(url);
+    });
+    if (!card) {
+      throw new Error('Should never get here');
+    }
+    // TODO capture additional deps from the compiled card
```

Comment:
This todo isn't super clear. What additional deps?

Reply:
@mattmcmanus so this would entail recursing through the `adoptsFrom` and the `fields` properties of the `CompiledCard` to record all the cards that we encounter. For example, if we have an `author` field that uses a `Person` card and the `Person` card itself adopts from a `User` card, then `User` is a dependency of our card. Like that kind of stuff...

Reply:
This is basically @ef4 's first comment on the PR.

Reply:
I made a note above that I think we really need to have a card service update in place before I can really write this kind of test. it would be the sort of thing were we _update_ one of our deps and we should observe that triggers an invalidation of our card. we can really only push so far with purely a _create_ mutator in the card service and see deep invalidations, i think this PR is the extent of that. So I think we'll have to circle back and write those tests after the update is ready.